### ISSUE TITLE
Loading pages takes too long

### ISSUE THREAD

**Issue by x011** [MrAlex94/Waterfox#2995]
### What happened?

 Since a couple of weeks ago WF started taking a huge amount time to load pages, like 30 seconds or more. If I change to FF or Brave, the page loads immediately. This happens only randomly, most of the time everything is working fine.

I've been temporarily using FF, but I really like WF and want to understand why this is happening. How can I debug this properly?

### Reproducible?

- [x] I have tested this on Firefox.

### Version

G5

### What platform are you seeing the problem on?

Windows

**Comment by arleas** [comment]
I'm having the same problem but it's only the first website I try.  After that, I can browse normally.  Cached webpages will seem to load quickly but if I click on any link the whole page goes white and it takes 20-30 seconds to wake up.   I initially thought this was an issue with pihole but it turned out that chrome and firefox didn't have this issue at all.  If I try to go to the pihole admin webpage either it's a white page for 30 seconds and then it pops up (and this is a webpage on my local network), OR the webpage pops up instantly but everything acts like it's trying to refresh all the data showing a circular rotating arrow instead of the graphs it normally shows.

This actually started maybe one or two updates ago from the latest version.   If I go do something for a while and come back to waterfox still open, it then acts like it does on first load.  If I close waterfox after using it but open it right back up, it's fine, but if I close it for a while and open it up again it acts like it does on first load.  I'm not sure how long I have to wait before this happens, but probably at least 5 minutes (but I haven't tested it)

**Comment by x011** [comment]
This may be related to windows 10/11 efficiency mode. On Firefox, you can disable it by going to `about:config` and setting `dom.ipc.processPriorityManager.backgroundUsesEcoQoS` to `False`.
I'll report back if this actually solves the problem as I'm currently testing it.

**Comment by arleas** [comment]
Well if that's the case, then it should also appear on Firefox as well but it doesn't (at least not for me).  I just looked at Firefox (since I uninstalled Waterfox for the time being) and saw that the about:config item you mentioned was still enabled on Firefox but I haven't noticed a problem.  I just reinstalled Waterfox to test it and sure enough, right out of the gate it basically acted like it froze up for 30 seconds.  

![pihole admin page](https://user-images.githubusercontent.com/53341110/229222504-f31fb0b9-ea59-4b0c-af50-ed55168e64dd.png)

I couldn't even begin to type my password until waterfox decided to wake up.  If I had already been logged in, it would have allowed me to see the dashboard, but all the graphs would have a circular loading arrow on them while Waterfox decided to act up.   It would be odd to me that a setting in the about:config would cause an issue on Waterfox while the same config on Firefox did not hinder me at all.

**Comment by x011** [comment]
Setting `dom.ipc.processPriorityManager.backgroundUsesEcoQoS` to `False` didn't solve the problem for me. 
Also, after using Firefox for about 2 weeks, this started to happen, but the _lagging_ is lower than in waterfox.

**Comment by x011** [comment]
I hope not, but I start believing this project was abandoned because there are no replies to bug reports for months already.
I switched back to FF and it fixed several WF bugs, specially passwords not being automatically filled and the time it takes to load pages decreased.

**Comment by arleas** [comment]
Yeah.  Sadly it's starting to remind me of Pale Moon.  Good browser until it wasn't anymore.   At least the mobile version of Waterfox still works well enough that I'm still using it. It's just the desktop windows 10 version that's messed up right now.  I also have tried LibreWolf but for now I'm just using plain Firefox until Waterfox can get it together.
